docs/en/stack/ml/nlp/ml-nlp-shared.asciidoc

```diff
@@ -1,3 +1,22 @@
+tag::ml-nlp-adaptive-allocations[]
+The numbers of threads and allocations you can set manually for a model remain constant even when not all the available resources are fully used or when the load on the model requires more resources.
+Adaptive allocations, however, provide full control and flexibility over the model allocations, and helps you to manage performance and cost more easily.
+When adaptive allocations is enabled, the number of allocations of the model is set automatically based on the current load.
+When the load is high, a new model allocation is automatically created.
+When the load is low, a model allocation is automatically removed.
+
+You can enable adaptive allocations by using:
+
+* the Create inference endpoint API for {ref}/infer-service-elser.html[ELSER], {ref}/infer-service-elasticsearch.html[E5 and models uploaded through Eland] that are used as {infer} services.
+* the {ref}/start-trained-model-deployment.html[start trained model deployment] or {ref}/update-trained-model-deployment.html[update trained model deployment] APIs for trained models that are uploaded to and managed on {ml} nodes.
```

Comment:
> for trained models that are uploaded to and managed on {ml} nodes.

@jan-elastic I need some input for phrasing this more accurately.

Reply:
I would replace "are uploaded to and managed on" by "are deployed on".